src/test/unit/osd_unittest.cc

```diff
@@ -0,0 +1,568 @@
+    displayPortTestBufferSubstring(2, 8,  "END BATTERY       : 15.2%c", SYM_VOLT);
+    displayPortTestBufferSubstring(2, 9,  "MIN RSSI          : 25%");
+    displayPortTestBufferSubstring(2, 10, "MAX ALTITUDE      :  6.5%c", SYM_FT);
+}
+
+/*
+ * Tests the calculation of statistics with metric unit output.
+ * (essentially an abridged version of the previous test
+ */
+TEST(OsdTest, TestStatsMetric)
+{
+    // given
+    // using metric unit system
+    osdConfigMutable()->units = OSD_UNIT_METRIC;
+
+    // and
+    // default state values are set
+    setDefualtSimulationState();
+
+    // when
+    // the craft is armed
+    doTestArm();
+
+    // and
+    // these conditions occur during flight (simplified to less assignments than previous test)
+    rssi = 256;
+    GPS_speed = 800;
+    GPS_distanceToHome = 100;
+    simulationBatteryVoltage = 147;
+    simulationAltitude = 200;
+    simulationTime += 1e6;
+    osdRefresh(simulationTime);
+    osdRefresh(simulationTime);
+
+    simulationBatteryVoltage = 152;
+    simulationTime += 1e6;
+    osdRefresh(simulationTime);
+
+    // and
+    // the craft is disarmed
+    doTestDisarm();
+
+    // then
+    // statistics screen should display the following
+    displayPortTestBufferSubstring(2, 3,  "ARMED TIME        : 00:02");
+    displayPortTestBufferSubstring(2, 4,  "FLY TIME          : 00:09");
+    displayPortTestBufferSubstring(2, 5,  "MAX SPEED         : 28");
+    displayPortTestBufferSubstring(2, 6,  "MAX DISTANCE      : 100%c", SYM_M);
+    displayPortTestBufferSubstring(2, 7,  "MIN BATTERY       : 14.7%c", SYM_VOLT);
+    displayPortTestBufferSubstring(2, 8,  "END BATTERY       : 15.2%c", SYM_VOLT);
+    displayPortTestBufferSubstring(2, 9,  "MIN RSSI          : 25%");
+    displayPortTestBufferSubstring(2, 10, "MAX ALTITUDE      :  2.0%c", SYM_M);
+}
+
+/*
+ * Tests activation of alarms and element flashing.
+ */
+TEST(OsdTest, TestAlarms)
+{
+    // given
+    // default state is set
+    setDefualtSimulationState();
+
+    // and
+    // the following OSD elements are visible
+    osdConfigMutable()->item_pos[OSD_RSSI_VALUE]        = OSD_POS(8, 1)  | VISIBLE_FLAG;
+    osdConfigMutable()->item_pos[OSD_MAIN_BATT_VOLTAGE] = OSD_POS(12, 1) | VISIBLE_FLAG;
+    osdConfigMutable()->item_pos[OSD_FLYTIME]           = OSD_POS(1, 1)  | VISIBLE_FLAG;
+    osdConfigMutable()->item_pos[OSD_ALTITUDE]          = OSD_POS(23, 7) | VISIBLE_FLAG;
+
+    // and
+    // this set of alarm values
+    osdConfigMutable()->rssi_alarm = 20;
+    osdConfigMutable()->cap_alarm  = 2200;
+    osdConfigMutable()->time_alarm = 1; // in minutes
+    osdConfigMutable()->alt_alarm  = 100; // meters
+
+    // and
+    // using the metric unit system
+    osdConfigMutable()->units = OSD_UNIT_METRIC;
+
+    // when
+    // the craft is armed
+    doTestArm(false);
+
+    // then
+    // no elements should flash as all values are out of alarm range
+    for (int i = 0; i < 6; i++) {
+        simulationTime += 1e6;
```

Comment:
isn't 1s delay to much? It may easily alias with blink frequency (which is usually expressed in Hz)